modules/matchPattern.js

```diff
@@ -1,22 +1,20 @@
 import pathToRegexp from 'path-to-regexp'
 
-const cache = {}
+// cache[exactly][pattern] contains getMatcher(pattern, exactly)
+const cache = {true: {}, false: {}}
```

Comment:
You can add a small ternary to use proper string keys. To help with that, you can store that as a separate variable, `cache_type` for instance.